### PR TITLE
feat(taskflow): emit code-level completion notification on task submission

### PR DESCRIPTION
**Pull request by LUOSENGWA**
## What

TeamHarness taskflow: **code-level completion notification + lifecycle attention events** (v2, design issue #1229).

Worker completion events used to be prompt-dependent (the LLM hand-sending a Matrix message; "a tool call does not count as the completion message" was unenforceable). This PR makes task-lifecycle attention signals first-class, idempotent, auditable room events generated by code:

1. **Completion events with a per-status contract** — `submit_task` now sends one first-line token per result status (`TASK_COMPLETED` / `TASK_PARTIAL` / `TASK_REVISION_NEEDED` / `TASK_BLOCKED` / `TASK_FAILED`), so leader-side prompts can branch on the line itself. The submitted status is validated against the accepted set; unknown values are rejected with a clear error.
2. **`@initiator` routing** — human team members (the task initiator) are `@mentioned` alongside the leader in completion, attention, and project events, closing the "requester only gets ambient room" gap (#1229 Part A, Q2).
3. **P0 ordering: sync before notify** — the completion event is an attention signal, not a receipt. Both `submit_task` and `complete_project` follow the same order: local state is persisted first, shared storage is synced, and only then is the event sent. A failed sync **withholds the notification** and returns a retryable failure (the local task/project state is already `submitted`/`completed`, so the retry is idempotent and the event is sent exactly once). A leader can no longer be woken by an event whose artifacts or state it cannot read.
4. **`request_attention` (new taskflow action)** — in-flight human decisions (`kind`: approval / decision / escalation / other) as first-class events: `ATTENTION_<KIND>: <task-id> - <question>`, mentions leader + humans, idempotent per unresolved kind (retry reuses the recorded event), terminal tasks rejected, sync-first like submit. Resolved by `accept_task_result` (closes all outstanding attention on the task) or an explicit `resolved: true`. Roles: worker / leader / remote-member (#1229 Q1).
5. **`PROJECT_COMPLETED` on `complete_project`** — a finished project now wakes the room (first task room, falling back to the project source room when it is a Matrix room) with an idempotent event. `projectCompletionEventId` is persisted only after a successful send; a retried `complete_project` reuses the recorded event.

**Idempotency**: transaction ids are stable per (task, status) / (task, kind, attempt) / (project, status); recorded event ids are persisted in task/project state so retries reuse instead of duplicate. A resubmission with a **changed** status invalidates the recorded pair and sends a fresh event (fixes the silent-resubmit-reuse gap). Pre-upgrade tasks (no recorded status) keep the old reuse behavior.

**Best-effort** (unchanged from v1): notification-level failures (no room / leader / Matrix env, membership missing, HTTP error) never block the state mutation; the `notificationNeeded` hint is kept for the requester reply-route report.

### Verification

- Full `test-taskflow.rb` harness green locally: all pre-existing assertions + 13 new groups — P0 ordering for `submit_task` (fail → withhold → idempotent retry) and the new P0 ordering for `complete_project` (forced project-dir sync failure → no `PROJECT_COMPLETED` event, no `projectCompletionEventId` recorded, retryable failure; the retry after recovery sends the event exactly once and persists the event id), per-status tokens, `@initiator` mentions, status validation, changed/same-status resubmit, `request_attention` lifecycle (send / idempotent / different kind / explicit close / terminal guard / accept-resolves-all), `PROJECT_COMPLETED` send + retry reuse.
- Go controller: all 21 packages green. Plugin gates: Ruby 9/9, plugin pytest 21 passed, CLI / validate / package green. Pre-existing failures in this environment (copaw test suite, CLI loongsuite probe contract) verified identical on the base commit — unrelated to this change.

### Follow-ups (noted, not in this PR)

- DM salience step for attention events (v1 delivers room `@mentions` only; a dedicated DM to humans is recorded as follow-up in #1229).
- Mention filtering for humans not in the room is intentionally not implemented: Matrix semantics already prevent non-members from seeing room events, so the extra check is a no-op.

### Related

- Design issue: #1229 (Part A: task lifecycle attention events & channel routing)
- PR #1219 (worker channel configuration proxy — separate scope)

---

TeamHarness taskflow：**代码级完成通知 + 生命周期 attention 事件**（v2，设计 issue #1229）。

Worker 完成事件原依赖 prompt（LLM 手动发 Matrix 消息，"工具调用不算完成消息"不可强制）。本 PR 把任务生命周期的 attention 信号变成代码生成的一等、幂等、可审计的房间事件：

1. **每状态一条契约首行**——`submit_task` 按结果状态发送 `TASK_COMPLETED` / `TASK_PARTIAL` / `TASK_REVISION_NEEDED` / `TASK_BLOCKED` / `TASK_FAILED` 首行 token，Leader prompt 可直接按行分支。提交的状态对 accepted 集合做校验，非法值明确报错。
2. **`@initiator` 路由**——人类成员（任务发起人）与 Leader 一起在完成 / attention / 项目事件中被 @，补上"发起人只拿到环境消息"的缺口（#1229 Part A Q2）。
3. **P0 顺序：先 sync 后 notify**——完成事件是 attention 信号不是回执。`submit_task` 与 `complete_project` 同一顺序：本地状态先落盘 → 共享存储 sync → 再发事件。sync 失败时**扣住通知**并返回 retryable 失败（本地状态已 submitted/completed，重试幂等、事件恰好一次）。Leader 不会再被"产物/状态不可读的事件"叫醒。
4. **`request_attention`（新 action）**——进行中的需要人类决策（approval / decision / escalation / other）变成一等事件：未解决同类幂等复用、终态任务拒绝、sync-first；由 `accept_task_result`（关闭该任务全部未决 attention）或显式 `resolved: true` 解决。角色：worker / leader / remote-member（Q1）。
5. **`complete_project` 发 `PROJECT_COMPLETED`**——项目完成唤醒房间（第一个 task room，回退项目 source room）；`projectCompletionEventId` 仅在发送成功后持久化，重试幂等复用。

**幂等**：txn 按 (task, status) / (task, kind, attempt) / (project, status) 稳定；事件 ID 落 task/project state，重试复用不重复；**状态变化**的重新提交作废旧事件并发新事件（修复静默复用缺口）；升级前任务保持旧复用行为。

**best-effort 不变**：通知层失败（无房间/Leader/Matrix 环境、成员校验、HTTP 错误）永不阻塞状态变更；`notificationNeeded` 保留给 requester reply-route。

**验证**：`test-taskflow.rb` 全量 harness 本地绿（含 13 组新断言，新增 complete_project P0 顺序：强制 project-dir sync 失败 → 无 `PROJECT_COMPLETED`、无事件 ID 落盘、retryable 失败；恢复后重试恰好发一次并持久化事件 ID）；Go 21 包全绿；插件门 Ruby 9/9 + pytest 21 过 + CLI/validate/package 绿；本环境预存失败（copaw 套件、CLI loongsuite 契约）与 base 逐一对平，与本改动无关。

**follow-up（不在本 PR）**：attention 事件的 DM 高显著度通道（v1 仅房间 @）；非房间成员 mention 过滤（Matrix 语义天然不可见，无需额外检查）。

### CODE REVIEW

File: plugins/teamharness/mcp/server.py [hunk 1]
```diff
@@ -4079,6 +4079,174 @@ def _send_delegate_notification(
...
+    if not room_id:
+        return {"sent": False, "skipped": True, "error": "task has no room_id"}
+    leader = _team_leader_matrix_id()
+    if not leader:
+        return {
+            "sent": False,
+            "skipped": True,
+            "error": "team leader Matrix ID not found in runtime config",
+        }
+    if task.get("completionEventId"):
```
Comment (oss-maintainer) [inline plugins/teamharness/mcp/server.py:4480]: **[Warning]** `completionEventId` reuse may swallow a *different* completion. `submit_task` leaves the task mutable (`"submitted"` is not in `TERMINAL_TASK_STATUSES`, and the plan node also becomes `submitted`), so the same `task_id` can be submitted again before the leader records a decision. Sequence: Worker submits `BLOCKED` → event persisted here → task later unblocked and resubmitted `SUCCESS` with the same `task_id` → this branch returns the stale BLOCKED event (`reused: true`) and no `TASK_COMPLETED` mention is ever sent — reproducing exactly the stall this PR fixes, just on a narrower path. Note the stable txn below (`submit-<task-id>`, line 4161) makes this structural: even without the reuse shortcut, Matrix would reject a re-PUT with a different body under the same txn id. Could you confirm whether a same-task_id resubmit after a BLOCKED submission is reachable in the runtime integration (Task Service resume flow)? If it is, consider keying the txn/event slot on `(task_id, result_status)` or an attempt counter, or invalidating `completionEventId` when the new `result_status` differs from the recorded one.

**Review by oss-maintainer — COMMENTED** [review]
### Summary
Mirrors the `delegate_task` notification pattern for `submit_task`: a code-level, stable-txn Matrix message with `m.mentions` to the leader, so completion wake signals no longer depend on the Worker LLM remembering the contract line. The design is sound and the implementation faithful to the existing pattern — verified against main@4ab6b15: all reused helpers exist (`_load_runtime_config`, `_section`, `_matrix_content`, `_canonical_room_id`, `_validate_assignee_membership`, `_write_task`), leader resolution normalizes roles the same way as `_roomflow_room_meta`, the membership guard prevents sending to users outside the room, failure is strictly best-effort (submission and artifact publishing never blocked), and the `notificationNeeded` requester hint is correctly preserved. The unused `arguments` parameter matches `_send_delegate_notification`'s existing signature style. Contract tests cover send/content, event-id persistence, retry reuse, the BLOCKED line, and forced-500 non-blocking. One warning inline: the `completionEventId` reuse branch can suppress a later, different completion (BLOCKED → same-task_id resubmit → SUCCESS), because both the reuse shortcut and the stable `submit-<task-id>` txn assume one terminal notification per task_id.

### Findings
- **[Warning]** server.py:4219 — completionEventId reuse + stable txn assume a single terminal notification per task_id; a same-task_id resubmit with a different result_status gets no new mention

---
*Automated review by github-manager-bot*

File: plugins/teamharness/mcp/server.py [hunk 2]
```diff
@@ -4079,6 +4079,174 @@ def _send_delegate_notification(
         return {"sent": False, "error": f"Matrix API error: {exc}"}
 
 
+def _team_leader_matrix_id() -> str:
+    """Resolve the team leader's Matrix user ID from the runtime config.
+
+    The controller projects the full team roster (with roles and Matrix
```
Comment (oss-maintainer) [inline plugins/teamharness/mcp/server.py:4124]: `_send_task_completion_notification` accepts an `arguments` parameter but never reads it. The orchestrator `_task_completion_notification` passes it through, but the send function only uses its keyword arguments. Consider dropping the parameter (or documenting why it is reserved) to avoid dead-code confusion.

File: plugins/teamharness/mcp/server.py [hunk 3]
```diff
@@ -4079,6 +4079,174 @@ def _send_delegate_notification(
...
+        task_id=task_id,
+        status=status,
+        summary=summary,
+        leader=leader,
+        worker=str(task.get("assigned_to") or ""),
+        result_path=str(task.get("result_path") or ""),
+    )
+    if notification.get("sent"):
+        task["completionEventId"] = notification.get("eventId")
+        _write_task(arguments, task)
```
Comment (oss-maintainer) [inline plugins/teamharness/mcp/server.py:4521]: If `_write_task` raises after the Matrix PUT succeeded, `completionEventId` is not persisted. The stable txn ID `submit-<task-id>` prevents a true duplicate in Matrix on retry, but the retry will make a redundant HTTP round-trip and return `reused: false` instead of `reused: true`. Consider persisting `completionEventId` before the HTTP call (optimistic write) or catching `_write_task` errors explicitly so the response reflects the actual state.

File: plugins/tests/teamharness/mcp/tools/test-taskflow.rb [hunk 4]
```diff
@@ -410,6 +426,56 @@ def block_yaml_import(name, *args, **kwargs):
...
+        raise AssertionError(f"submit_task did not persist completionEventId: {submitted_meta!r}")
+    if submitted_meta.get("completionEventId") != submitted.get("notification", {}).get("eventId"):
+        raise AssertionError(f"persisted completionEventId mismatch: {submitted_meta!r} vs {submitted.get('notification')!r}")
+
+    resubmitted = payload("taskflow", {
+        "role": "worker",
+        "action": "submit_task",
+        "payload": {
+            "taskId": task_id,
+            "status": "SUCCESS",
```
Comment (oss-maintainer) [inline plugins/tests/teamharness/mcp/tools/test-taskflow.rb:468]: Tests cover SUCCESS, BLOCKED, retry, and HTTP-failure paths well, but the skip branches are untested: no-leader-in-config (standalone run), leader-not-in-room (membership guard), and no-Matrix-env. These are documented best-effort paths, but a missing leader or missing env var are common deployment configurations worth a smoke assertion (e.g., `notification.skipped == true`).

**Review by oss-maintainer — APPROVED** [review]
### Summary
This PR adds automatic Matrix completion notifications to submit_task, mirroring the existing delegate_task pattern with stable transaction IDs for idempotency. The implementation is correct, best-effort by design (notification failures never block submission), and well-tested across success, retry, BLOCKED, and HTTP-failure paths. All findings are informational: an unused parameter, an edge case where _write_task failure could cause a redundant HTTP call on retry (but not a duplicate message), and missing test coverage for skip branches (standalone mode, membership failures). The change is backward-compatible (additive response field, preserves notificationNeeded hint) and solves a real production issue where workers forget to send completion mentions after context compaction.

---
*Automated review by github-manager-bot*

**Comment by shiyiyue1102** [comment]
The two newly addressed `resolved: true` cases look correct, including the sync-failure/retry coverage. One idempotency case remains:

When a new `request_attention` is persisted locally but its first `_sync_task` fails, the unresolved record has no `eventId`. A retry with the same kind skips the reuse branch (which requires `eventId`), appends a second unresolved record with a new attempt/transaction id, and can later send a second attention event. This contradicts the retryable/idempotent contract after a failed sync.

Please reuse the pending same-kind record even before it has an event id, retry its sync and then send/persist one event; add a forced first-sync-failure → recovery retry test that proves there is one record and one room event. Once this is fixed and the current CI is green, I will merge.

---

本轮 `resolved: true` 的两个问题已修复，且同步失败/重试覆盖也正确；但还有一个幂等场景：

新的 `request_attention` 已写入本地、但首次 `_sync_task` 失败时，这条未解决记录还没有 `eventId`。同 kind 重试会跳过要求 `eventId` 的复用分支，新增第二条未解决记录及新的 attempt/事务 ID，恢复后可能发送第二个 attention 事件。这不符合“同步失败后可重试且幂等”的契约。

请在记录尚未获得 event id 时也复用同 kind 的 pending 记录：重试其同步，成功后只发送并持久化一个事件；并补充“首次同步失败 → 恢复重试”测试，断言最终只有一条记录和一个房间事件。修复后且当前 CI 全绿，我会立即合并。

**Comment by LUOSENGWA** [comment]
@shiyiyue1102 

Both `request_attention` lifecycle gaps are fixed in `e79a5191`:

1. **Existing-record close is now sync-first** (same shape as the approved `submit_task` / `complete_project` ordering). The close marks the record resolved locally, pushes the task dir via `_sync_task`, and only then reports success. A failed close-sync returns `ok: false` / `retryable: true` / `statePersisted: true` with **no event sent**, and the error names the exact state: the shared `meta.json` may still show the loop open, retry with the same payload. One subtlety worth flagging: after a failed close the local record is *already* resolved, so a naive retry would find no open record and fall through to the new-record path (a phantom record + a new ping — the same bug class). The close handler therefore targets (a) the most recent **unresolved** same-kind record, or (b) if all same-kind records are resolved, the most recent one **with an eventId** — re-affirming it is idempotent: re-write, re-sync, reuse the original event, no new ping, no new record. The retry converges on the first sync that succeeds.
2. **First-call `resolved: true` is rejected** (chosen over no-op). There is no open loop to close, and a no-op success would silently swallow a caller that meant to raise: it would believe the loop closed while the decision it needed never surfaced. The call now returns the structured error `no attention record of kind '<kind>' to resolve; call request_attention without resolved to raise one` — nothing is recorded, no event is sent. The new-record path also pins `resolved: false` explicitly, since an explicit resolve can no longer reach it.
3. **Regression tests** (`test-taskflow.rb`, reusing the existing `TEAMHARNESS_TEST_FAIL_SYNC_TASK` hook): (a) close under a forced sync failure → `ok: false` / `retryable: true` / `synced: false`, still exactly one event on the wire, and the `mc` log delta proves the mirror push was *attempted* (the pre-fix code never called `_sync_task` on this branch, so the log delta is the discriminating assertion); (b) the idempotent retry after recovery → `ok: true` / `synced: true`, the shared `meta.json` shows the record **resolved**, the original event id is reused, exactly one record of that kind, no second event; (c) first-call `resolved: true` with no same-kind record → rejected, nothing recorded, zero events. Verified **red against the pre-fix code** (the old branch returned `ok: true` / `synced: true` for the failed close), green after.
4. **Full sweep of the other write→sync→notify paths** (so this does not become a third round): delegate (sync-gated before notify, retryable), submit first + resubmit (sync-gated, approved), complete_project (sync-gated, approved), cancel (sync-gated), accept_task_result's attention auto-close (inside the task sync gate; a retry repairs an earlier locally-committed state), attention new-record (sync-gated; the post-send `eventId` persist is local-only, same documented precedent as `completionEventId` — the stable txn `attention-<task>-<kind>-<attempt>` makes any retry deduplicate at the homeserver). `ack_task` and `create_quick_project` report `synced` without a failure gate, but they are pre-existing, idempotent, and self-healing (a retry re-syncs; the leader-side `check_task` pulls shared state), and no caller depends on the shared copy of that state — intentionally unchanged, noted here for completeness. No other path reports success over shared state without syncing it.
5. **Docs**: the design doc's `request_attention` section gains the close contract (sync-first close, retry convergence, first-call rejection), item 10 of the numbered contract list mirrors it, and the tool schema + task-execution skill document the close semantics for LLM callers.

Local gates: taskflow harness green including the new block (498 → 518 `mc` invocations, all ordered as asserted), the other 8 Ruby suites green (`test-projectflow` fails identically on the base commit — pre-existing, `accept_task_result requires leader role`), pytest 141 passed with the same 2 pre-existing `test_trace_integration` failures as the base. CI note from the 09-14 comment stands: the integration suite's `paths:` filter does not cover this PR's files, so review remains the gate; happy to push a no-op re-trigger commit if you want a fresh full run.

---
`request_attention` 的两条生命周期缺口已在 `e79a5191` 修复：

1. **已有记录的 close 现在 sync-first**（与已认可的 `submit_task` / `complete_project` 顺序同形）。close 先把记录标记 resolved 落本地、`_sync_task` 推任务目录、**成功后**才报成功；同步失败返回 `ok: false` / `retryable: true` / `statePersisted: true` 且**不发任何事件**，错误文案点明共享 `meta.json` 可能仍显示未解决、用相同 payload 重试。一个值得指出的细节：close 失败后本地记录**已经**是 resolved，naive 重试会找不到 open 记录而掉进新记录路径（幻影记录 + 新 ping——同类 bug）。因此 close 处理的目标是：(a) 最近的**未解决**同类记录；(b) 若同类记录已全部 resolved，取最近一条**带 eventId** 的记录重新确认——重写、重新同步、复用原事件、无新 ping、无新记录，幂等收敛到首次同步成功。
2. **首次调用 `resolved: true` 直接拒绝**（在"拒绝 / no-op"两选项中选了拒绝）。没有 open loop 可关；no-op 成功会静默吞掉调用方本意是"发起"的意图——它会以为闭环已关，而它需要的人工决策根本没浮出。现在返回结构化错误 `no attention record of kind '<kind>' to resolve; call request_attention without resolved to raise one`——不落任何记录、不发任何事件。新记录路径显式钉死 `resolved: false`（explicit resolve 已不可能到达该路径）。
3. **回归测试**（`test-taskflow.rb`，复用现有 `TEAMHARNESS_TEST_FAIL_SYNC_TASK` hook）：(a) 强制同步失败下的 close → `ok: false` / `retryable: true` / `synced: false`、线上仍恰好 1 个事件、`mc` 日志增量证明 mirror push **被尝试过**（修复前代码该分支从不调 `_sync_task`，日志增量即判别断言）；(b) 恢复后幂等重试 → `ok: true` / `synced: true`、共享 `meta.json` 中记录已 **resolved**、复用原事件 ID、该 kind 恰好 1 条记录、无第二事件；(c) 无同类记录的首次 `resolved: true` → 被拒、零记录、零事件。已对**修复前代码验证过红**（旧分支对失败 close 返回 `ok: true` / `synced: true`），修复后全绿。
4. **其余 write→sync→notify 路径全量扫描**（避免再来第三轮）：delegate（sync 门控在前、可重试）、submit 首提 + 重提（门控、已认可）、complete_project（门控、已认可）、cancel（门控）、accept_task_result 的 attention 自动关闭（在 task sync 门内；重试可修复先前本地已提交的遗留状态）、attention 新记录（门控；发送后的 `eventId` 落本地为 best-effort，与 `completionEventId` 同一文档先例——稳定事务 ID `attention-<task>-<kind>-<attempt>` 让任何重试在 homeserver 去重）。`ack_task` 与 `create_quick_project` 报 `synced` 但不设失败门控——预存行为、幂等、自愈（重试重新同步；Leader 侧 `check_task` 拉共享态），且无调用方依赖该共享副本，有意不改，此处完整说明。除此之外没有任何路径在"共享态未同步"时向调用方报成功。
5. **文档**：设计文档 `request_attention` 一节补 close 契约（sync-first close、重试收敛、首次调用拒绝），编号契约清单第 10 条同步，工具 schema 与 task-execution skill 对 LLM 调用方说明 close 语义。

本地门禁：taskflow harness 全绿（含新用例，`mc` 调用 498 → 518、顺序全符合断言），其余 8 个 Ruby 套件绿（`test-projectflow` 在 base commit 上同样失败——预存，`accept_task_result requires leader role`），pytest 141 过 + 与 base 相同的 2 个预存 `test_trace_integration` 失败。CI 说明沿用 09-14 评论：集成套件 `paths:` 过滤不覆盖本 PR 文件，review 即门禁；如需要新鲜全量 CI，我可以推一个 no-op re-trigger commit。

**Comment by LUOSENGWA** [comment]
@shiyiyue1102 

The pending-record idempotency case is fixed in `adefe4a9`:

1. **The reuse scan now catches the pending record even before it has an eventId.** A same-kind unresolved record is the one open loop of that kind, and there are now three cases instead of two: (a) it already has an `eventId` → plain reuse, no new ping (unchanged); (b) it is **pending** — created, but its first `_sync_task` failed so it carries no `eventId` → the retry **reuses it**: re-run `_sync_task` (a second failure returns the same retryable shape — no second record is appended, no event is sent, and the error now says the pending record is reused on retry), and on success send and persist **exactly one** event for it. The retry keeps the record's original `attempt` number, so the deterministic transaction id `attention-<task>-<kind>-<attempt>` is unchanged and any partial delivery still deduplicates at the homeserver; the notification uses the question recorded on the record, not a possibly-different retry payload. (c) no same-kind unresolved record → new-record path, byte-for-byte the same flow as before.
2. **Regression test** (`test-taskflow.rb`, forced first-sync-failure → recovery retry, reusing the `TEAMHARNESS_TEST_FAIL_SYNC_TASK` hook): the failed first attempt must be a retryable failure, must have attempted the mirror push (`mc` log delta), and must leave exactly one pending record (no `eventId`); the retry after recovery must succeed with `sent: true` + an `eventId`, leave **one** record of that kind carrying the event id, and produce **exactly one** room event; a third same-kind call must then hit the plain-reuse branch and still hold at one record / one event. **Negative control verified**: on the pre-fix code the retry reproduces exactly the case you described — two records (attempt 1 pending without `eventId`, attempt 2 carrying the new event id) — and the one-record assertion goes red; green after the fix.
3. **Docs**: the design doc's state bullet, the sync-first bullet, and item 10 of the numbered contract list now document the pending-record reuse semantics (one record, one event, retry never appends).

Local gates: taskflow harness green (518 → 535 `mc` invocations, all ordered as asserted), the other Ruby suites green, pytest 141 passed with the same 2 pre-existing `test_trace_integration` failures as the base. The two unit shards (`test`, `teamharness`) re-run on this commit; the integration-suite `paths:` note from 09-14 is unchanged.

---
pending 记录的重试幂等场景已在 `adefe4a9` 修复：

1. **复用扫描现在能捕获尚未获得 eventId 的 pending 记录。** 同 kind 的未解决记录就是该 kind 唯一 open loop，分支从两种扩成三种：(a) 已有 `eventId` → 普通复用、不发新 ping（行为不变）；(b) **pending**——记录已创建但首次 `_sync_task` 失败、尚无 `eventId` → 重试**复用该记录**：重跑 `_sync_task`（第二次失败返回同样的可重试形态——不追加第二条记录、不发事件，错误文案明确"pending 记录将在重试中复用"），成功后只为它**发送并持久化一个**事件。重试沿用记录的原始 `attempt` 号，确定性事务 ID `attention-<task>-<kind>-<attempt>` 不变，任何部分送达仍在 homeserver 去重；通知内容用记录上持久化的 question，而不是可能不同的重试 payload。(c) 无同 kind 未解决记录 → 走新记录路径，流程与之前逐字节一致。
2. **回归测试**（`test-taskflow.rb`，强制首次 sync 失败 → 恢复重试，复用 `TEAMHARNESS_TEST_FAIL_SYNC_TASK` hook）：首次失败必须返回可重试失败、必须尝试过 mirror push（`mc` 日志增量）、本地必须恰好留一条无 `eventId` 的 pending 记录；恢复后重试必须 `sent: true` + 带 `eventId` 成功、该 kind 只剩**一条**记录且携带事件 ID、房间事件恰好**一个**；第三次同 kind 调用必须命中普通复用分支，仍保持一条记录/一个事件。**负向对照已验证**：修复前代码精确复现你描述的场景——两条记录（attempt 1 pending 无 `eventId`、attempt 2 携带新事件 ID）——一条记录断言变红；修复后全绿。
3. **文档**：设计文档 state 条、sync-first 条、编号契约清单第 10 条已补 pending 记录复用语义（一条记录、一个事件、重试永不追加）。

本地门禁：taskflow harness 全绿（`mc` 调用 518 → 535、顺序全符合断言），其余 Ruby 套件绿，pytest 141 过 + 与 base 相同的 2 个预存 `test_trace_integration` 失败。两个 unit shard（`test`、`teamharness`）在本 commit 上重跑；09-14 的集成套件 `paths:` 说明不变。

**Review by shiyiyue1102 — APPROVED** [review]
LGTM. The task/project completion paths and request_attention now preserve sync-before-notify, retryable failure handling, and one-record/one-event idempotency across failed-sync recovery. The current required checks are green.

---

LGTM。任务/项目完成与 request_attention 现在均满足先同步后通知、同步失败可重试，以及失败恢复后的单记录/单事件幂等语义；当前检查均已通过。

**Comment by LUOSENGWA** [comment]
@shiyiyue1102 Thank you for your support, next #1158 , branch updated, CI is running.
